core/src/main/java/com/datastax/oss/driver/api/core/type/codec/registry/CodecRegistry.java

```diff
@@ -35,11 +36,39 @@
  * {@link Row#get(int, Class)}, {@link TupleValue#set(int, Object, Class)}, etc.)
  */
 public interface CodecRegistry {
+
+  // roughly sorted by popularity
+  TypeCodec<?>[] DEFAULT_PRIMITIVE_CODECS =
+      new TypeCodec<?>[] {
+        // Must be declared before AsciiCodec so it gets chosen when CQL type not available
+        TypeCodecs.TEXT,
+        // Must be declared before TimeUUIDCodec so it gets chosen when CQL type not available
+        TypeCodecs.UUID,
+        TypeCodecs.TIMEUUID,
+        TypeCodecs.TIMESTAMP,
+        TypeCodecs.INT,
+        TypeCodecs.BIGINT,
+        TypeCodecs.BLOB,
+        TypeCodecs.DOUBLE,
+        TypeCodecs.FLOAT,
+        TypeCodecs.DECIMAL,
+        TypeCodecs.VARINT,
+        TypeCodecs.INET,
+        TypeCodecs.BOOLEAN,
+        TypeCodecs.SMALLINT,
+        TypeCodecs.TINYINT,
+        TypeCodecs.DATE,
+        TypeCodecs.TIME,
+        TypeCodecs.DURATION,
+        TypeCodecs.COUNTER,
+        TypeCodecs.ASCII
+      };
+
```

Comment:
👍 , but I would move it to `DefaultCodecRegistry`: this is an implementation detail and I don't see a strong case for making it part of the public API.

Reply:
Believe me or not, It doesn't work. If `DEFAULT_PRIMITIVE_CODECS` is declared in `DefaultCodecRegistry` or `CachingCodecRegistry`,  then when you instantiate `DefaultCodecRegistry` from `CodecRegistry.DEFAULT`, if the class is not loaded yet the constructor will be invoked _before_ the static initializers, and as a consequence, `DEFAULT_PRIMITIVE_CODECS` will be null. That's why I moved `DEFAULT_PRIMITIVE_CODECS` to `CodecRegistry`. There must be some circularity going on. The only alternative would be to declare `DEFAULT_PRIMITIVE_CODECS` in a separate class.